src/ocean/net/device/Berkeley.d

```diff
@@ -1033,7 +1033,9 @@ deprecated public abstract class Address
                 hints.ai_flags = flags;
                 hints.ai_family = (flags & AIFlags.PASSIVE && af == AddressFamily.UNSPEC) ? AddressFamily.INET6 : af;
                 hints.ai_socktype = SocketType.STREAM;
-                int error = getaddrinfo(toStringz(host), service.length == 0 ? null : toStringz(service), &hints, &info);
+                int error = getaddrinfo((host ~ "\0").ptr,
```

Comment:
[We discussed it already](https://github.com/sociomantic-tsunami/ocean/pull/109#issuecomment-302082401).